### ISSUE TITLE
Unexpected keyword argument not detected in constructor

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#206]
Originally reported by: **the mulhern (BitBucket: [the_mulhern](http://bitbucket.org/the_mulhern))**

---

In the example below, pylint does not report E1123

```
#!python

class Junk(object):
    def __init__(self, fmt=None):
        self.__format = fmt

if __name__ == "__main__":
    Junk(dummy="%s is not cool")
```

in the slightly longer example below, it reports

 Passing unexpected keyword argument 'dummy' in function call (unexpected-keyword-arg)

for meth and meth2.

```
#!python

class Junk(object):
    def __init__(self, fmt=None):
        self.__format = fmt

    def meth(self, fmt):
        self.__format = fmt

    def meth2(self):
        pass

if __name__ == "__main__":
    junk = Junk(fmt="%s is not cool")
    junk.meth(dummy=2)
    junk.meth2(dummy=2)
```

It should report for the first example's constructor as well.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/206
